File: core/scripts/Java-Library/src/chh/math/set/EmptySet.java
package chh.math.set;
/**
 * <b>The Axiom of Existence:</b>
 * There exists a set that contains no elements; furthermore, this set is unique.
 * We symbolize this set by {}.
 * @author  <a href="mailto:chrisharvey2pi@gmail.com">Christopher H. Harvey</a>
 * @version 2013.12.08
 */
public class EmptySet extends Set {
	/**
	 * Constructs a new EmptySet object.
	 */
	public EmptySet() {}
	
	/**
     * {@inheritDoc}
	 * All EmptySet objects are empty by definition.
	 * @return false
     */
    @Override
	public final boolean contains(Set x) {
		return false;
	}
	/**
     * {@inheritDoc}
	 * The only possible subset of the empty set is itself.
	 * @return true if x is empty
     */
    @Override
	public final boolean includes(Set x) {
		return x.isEmpty();
	}
	/**
     * {@inheritDoc}
	 * All EmptySet objects are empty.
	 * @return true
     */
	@Override
	public final boolean isEmpty() {
		return true;
	}
	/**
	 * {@inheritDoc}
	 * @return false
	 */
	@Override
	public final boolean containsEmpty() {
		return false;
	}
	/**
     * {@inheritDoc}
	 * All EmptySet objects are empty.
	 * @return false
	 */
	@Override
	public final boolean isSingleton() {
		return false;
	}
	/**
     * {@inheritDoc}
	 * All EmptySet objects are empty.
	 * @return false
	 */
	@Override
	public final boolean isPair() {
		return false;
	}
	/**
     * {@inheritDoc}
	 * All EmptySet objects are empty.
	 * @return false
	 */
	@Override
	public final boolean isPowerSetOf(Set x) {
		return false;
	}
//	/**
//	 * {@inheritDoc}
//	 * All EmptySet objects are empty.
//	 * @return false
//	 */
//	@Override
//	public final boolean isSuccessorOf(Set x) {
//		return false;
//	}
	/**
     * {@inheritDoc}
	 * No EmptySet objects are inductive.
	 * @return false
	 */
	@Override
	public final boolean isInductive() {
		return false;
	}
	/**
     * {@inheritDoc}
	 * @return {@code {}} or {@code 0}
	 */
	@Override
	public String toString() {
		//return "{}";
		return "0";
	}
	
}

File: core/scripts/set/EmptySet.js
/**
  * There exists a set that contains no elements; furthermore, this set is unique.
  * We symbolize this set by '{}' or the symbol '0'.
  */
function EmptySet() {
  Set.call(this);
}
Util.extend(EmptySet, Set);

EmptySet.prototype.owns = function owns(x) {
  return false; // All EmptySet objects are empty by definition.
}

EmptySet.prototype.includes = function includes(x) {
  return x.isEmpty(); // The only possible subset of the empty set is itself.
}

EmptySet.prototype.equals = function equals(x) {
  return x.isEmpty(); // a convenience statement: all empty sets are equal
}

EmptySet.prototype.toString = function toString() {
  // return '{}';
  return '0';
}


EmptySet.prototype.isEmpty = function isEmpty() {
  return true;
}
EmptySet.prototype.ownsEmpty = function ownsEmpty() {
  return false;
}

EmptySet.prototype.isSingleton = function isSingleton() {
  return false;
}
EmptySet.prototype.isSingletonOf = function isSingletonOf(x) {
  return false;
}

EmptySet.prototype.isPair = function isPair() {
  return false;
}
EmptySet.prototype.isPairOf = function isPairOf(x, y) {
  return false;
}

EmptySet.prototype.isPowerSetOf = function isPowerSetOf(x) {
  return false;
}

EmptySet.prototype.isSuccessorOf = function isSuccessorOf(x) {
  return false;
}

EmptySet.prototype.isInductive = function isInductive() {
  return false;
}
